multinode/bootstrapping.md

```diff
@@ -0,0 +1,191 @@
+access node, this can be overridden using the `database` option to the
+`add_data_node` command.
+
+```
+SELECT * FROM add_data_node('data_node_1', 'data1.example.com',
+                            database => 'other');
+```
+
+The access node has to ensure that there is a properly configured
+database on the data node by either bootstrapping a new database or
+verifying that it is correctly set up, depending on the value of the
+`bootstrap` option to `add_data_node`. A data node can be added
+without bootstrapping by using the `bootstrap => false` option.
+
+```
+SELECT * FROM add_data_node('data_node_1', 'data1.example.com',
+                            boostrap => false);
+```
+
+If a data node is not bootstrapped, only the validation steps below
+are executed. If `bootstrap` is `true` (which is the default) the
+access node will:
+
+1. Ensure that there is a database on the data node.
+2. Ensure that the extension is available in the database.
+3. Set metadata on the data node.
+
+## Setting up the database on the data node
+
+The database name on the data node is either the name of the database
+on the access node, or the value of the `database` option to
+`add_data_node`.
+
+### Bootstrapping the database
+
+If `bootstrap` is true (which is the default), the access node will
+first connect to either the `postgres` database or, if the `postgres`
+database does not exist, to the `template0` database on the data node
+and create new database:
+
+```
+CREATE DATABASE <database>
+   TEMPLATE template0 OWNER <owner>
+   ENCODING <encoding> LC_COLLATE <collation>
+   LC_CTYPE <character type>
+```
+
+Here, the *owner* is the user connected on the access node, the
+*encoding*, *collation*, and *character type* parameters are taken
+from the current database on the access node.
+
+Once the database is created, the access node disconnect and continue
+with bootstrapping the extension. If the database already exists on
+the data node, the procedure proceeds with setting up the extension.
+
+[Currently, there is no validation of the database as there is if
+`bootstrap` is false. IMHO, we should add that. /Mats]
+
+### Validation of the database
+
+If `bootstrap` is false, the database will be validated by checking
+that the encoding, character type, and collation are the same as the
+current database on the access node.
+
+## Setting up the extension
+
+The `timescaledb` extension needs to be installed in the database on
+the data node, so the next step is ensuring that there is a valid
+extension installed in the database by either: Creating the extension,
+or verifying that the extension is compatible with the extension
+installed on the access node.
+
+### Creating the schema for the extension
+
+To install the extension, a schema is created to store the objects
+that the `timescaledb` extension needs. The schema name is decided
+looking what schema is used for the extension on the access node and
```

Comment:
If the extension is installed in several schemas on the access node, which one is it going to be used? Is it the same schema where the called `add_data_node` is located? May be provide an example.

E.g.:
```sql
CREATE SCHEMA distributed_db;
CREATE EXTENSION timescaledb SCHEMA distributed_db;
SELECT distributed_db.add_data_node(...);
```

Reply:
```suggestion
looking at what schema name is used for the extension on the access node and
```